core-plugins/src/main/java/co/cask/hydrator/plugin/transform/ProjectionTransform.java

```diff
@@ -316,6 +336,8 @@ private Schema getOutputSchema(Schema inputSchema) {
       String inputFieldName = inputField.getName();
       if (fieldsToDrop.contains(inputFieldName)) {
         continue;
+      } else if (!fieldsToKeep.isEmpty() && !fieldsToKeep.contains(inputFieldName)) {
```

Comment:
The right thing would be throw an exception during configure time, this will throw an error during deployment of the pipeline. This will be applied not only to keep, but also to drop and rename functionalities. @riwazp7 can you open a JIRA for schema validation for projection transform?